### PR TITLE
Fixes #8183 Remove separate color for digit 0 to maintain uniformity

### LINKED ISSUE #8183: Digits not very visible in the night mode[Bug]
###### Reproduction Steps

1. Go to the navigation drawer
2. switch to night mode
3. Check the last 2 digits (when 0) on the right side of a deck
ankil
ankid

(This is more of an enhancement than a bug)
###### Expected Result
A lighter color which is more visible and suitable for night mode

###### Actual Result
The digits are not very visible in a dark theme

###### Debug info
AnkiDroid Version- latest version (from the master branch)
Android version-10

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by 2tanayk**
## Purpose / Description
The digit 0 was given a different color on the digits on the deck picker(for the new, learning and to review digits) and this was not very visible for the night mode and further the aforementioned thing did not reflect for the same digits on opening a deck. Hence a different color scheme for digit 0 was removed in this PR.

## Fixes
Fixes #8183 

## Approach
_How does this change address the problem?_
I've removed `mZeroCountColor` from being assigned as the color of a digit and instead let them be the same that is: `mNewCountColor `, `mLearnCountColor `, `mReviewCountColor` for the respective digits.
## How Has This Been Tested?
This has been tested on an actual device running on Android 10.

## Screenshots
anki1
anki2
anki3
anki4

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by mrudultora** [comment]
Hi @2tanayk you have added a lot of additional whitespaces in your code. 

>  You have not changed whitespace unnecessarily (it makes diffs hard to read)

File: AnkiDroid/src/main/java/com/ichi2/anki/widgets/DeckAdapter.java [hunk 1]
```diff
@@ -94,6 +96,8 @@
     // Whether we have a background (so some items should be partially transparent).
     private boolean mPartiallyTransparentForBackground;
 
+
```
Comment (mrudultora) [inline AnkiDroid/src/main/java/com/ichi2/anki/widgets/DeckAdapter.java:99]: These extra whitespace lines are not really needed. There are many such lines in this PR. Please remove them ;-)

**Review by mrudultora — CHANGES_REQUESTED** [review]
Hi, there are some basic things ( we must not add additional whitespaces unnecessarily) to change.

**Comment by 2tanayk** [comment]
Not unnecessary ones I think,what should I change?

**Comment by mrudultora** [comment]
> Not unnecessary ones I think,what should I change?

You have added so many blank lines which are not at all needed. (for ex, the changes I suggested please view those lines). In some parts of your code, there are 3 blank lines between the code. This is not needed. You got my point ? 
Wait just click Files changed in the above bar (below the title of this PR) and see all the green lines. Green lines are added by you. Then, you will get my point.

**Comment by 2tanayk** [comment]
> > Not unnecessary ones I think,what should I change?
> 
> You have added so many blank lines which are not at all needed. (for ex, the changes I suggested please view those lines). In some parts of your code, there are 3 blank lines between the code. This is not needed. You got my point ?
> Wait just click Files changed in the above bar (below the title of this PR) and see all the green lines. Then, you will get my point.

Yeah I do know that,I thought that won't be an issue,ok I'll fix it

**Comment by mikehardy** [comment]
We may need to think a little more closely here - while it is true this could be more visible the 0 is almost invisible on purpose from Anki Desktop I think, and this would be a deviation from Anki Desktop, which is not something we do normally unless specifically required for the mobile form factor (which this is not)

Has then been checked with Anki Desktop? Perhaps if we are interested in Accessibility improvements we should start there first so we don't deviate, and the whole ecosystem benefits?

**Comment by 2tanayk** [comment]
image

This is how Anki Desktop looks with night mode,so what do we do next?

**Comment by Arthur-Milchior** [comment]
Ther is indeed a purpose for the 0 to be of a specific color. The goal is that, when you are looking for decks to review, you can see in a fraction of seconds the decks which still have card. 
So, using the same color everywhere is not a good choice for me.

Following Anki desktop is also a bad choice because it is not accessible. We must remain very close to Anki's logic, I don't think that changing the color is a huge problem.

So I see only two acceptable solutions:
* removing the 0s entirely
* Keeping a shade of grey, but one that is satisfies usual readability constraints.

The logic being that, etiher we want everybody to be able to read it, or nobody. But restricting it to person with good view makes no sens. 

I'd choose keeping the 0s. This would have the added advantages that, if at some point people are able to use the app with a screen reader (not really possible now I fear), they will always hear three numbers. Otherwise, if you hear "Biology 2", you don't know in which column the two is

**Comment by 2tanayk** [comment]
ok so to satisfy readability we choose a color right? and one that passes the accessibility test

**Comment by eginhard** [comment]
Some accessibility aspects will be hard to get right without consulting end users and only following the guidelines of an automatic tool that has limited options for configuration. Unfortunately I can't help with that either, but here are some thoughts on this.

The grey zeros don't carry much information, they only mark the absence of cards to study. All grey numbers are zeros, there is never a need to look at them in detail. While it might take some effort the first time around, this should not impede regular use of the app. In fact, them having a lower contrast in this table format makes it much easier to read the coloured numbers that actually do carry information as Arthur pointed out. Note that in the more detailed deck statistics screen (when you click on the numbers) zeros are shown in the regular colours. 

I agree that the zeros should remain and that there needs to be some adjustment to the contrast, especially in night mode, but a 4.5 contrast ratio might not be needed. The WCAG also have a provision that excludes "incidental" text (e.g. text that is only decorative or on inactive elements) that might be applicable here to some degree: https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html

**Comment by mikehardy** [comment]
I think the zeros are "inactive elements" yes.
Also, I think following Anki visually is important, not to obstruct accessibility but to drive change upstream as well. 
I don't see a reason to change the zeros, personally :shrug:

**Comment by mikehardy** [comment]
I think this is resolved -> wontfix however, learning that in *some* cases ("inactive elements") it is okay to violate normal accessibility contrast rules is quite useful. This should be used sparingly (and I think we use it sparingly...) but I do think it applies here.

**Pull request closed without merging**